Mage/src/main/java/mage/abilities/mana/ManaOptions.java

```diff
@@ -61,34 +62,20 @@ public void addMana(List<ActivatedManaAbilityImpl> abilities, Game game) {
 
         } else { // mana source has more than 1 ability
             //perform a union of all existing options and the new options
-            List<Mana> copy = new ArrayList<>(this);
-            this.clear();
+            ManaOptions out = new ManaOptions();
             for (ActivatedManaAbilityImpl ability : abilities) {
                 for (Mana netMana : ability.getNetMana(game)) {
                     checkManaReplacementAndTriggeredMana(ability, game, netMana);
                     for (Mana triggeredManaVariation : getTriggeredManaVariations(game, ability, netMana)) {
-                        SkipAddMana:
-                        for (Mana mana : copy) {
-                            Mana newMana = new Mana();
-                            newMana.add(mana);
-                            newMana.add(triggeredManaVariation);
-                            for (Mana existingMana : this) {
-                                if (existingMana.equalManaValue(newMana)) {
-                                    continue SkipAddMana;
-                                }
-                                Mana moreValuable = Mana.getMoreValuableMana(newMana, existingMana);
```

Comment:
There were a more complex logic -- ignore duplicated mana option, ignore useless mana option. New code will add any mana option to the list.

New simplified code must be a better solution cause it run mana optimization anyway, but it must be researched. Too much useless mana options (that can't be removed after optimization) can generate bad performance and memory overflow. So need additional research.

Reply:
I'll look at it later